service/src/main/java/io/camunda/service/DocumentServices.java

```diff
@@ -83,49 +81,47 @@ public CompletableFuture<DocumentReferenceResponse> createDocument(
     final List<Either<DocumentErrorResponse, DocumentReferenceResponse>> results =
         new ArrayList<>();
 
-    final List<CompletableFuture<Void>> futures =
+    final List<CompletableFuture<Boolean>> futures =
         requests.stream()
             .map(
                 request -> {
                   final var storeRequest =
                       new DocumentCreationRequest(
                           request.documentId, request.contentInputStream, request.metadata);
-                  return getDocumentStore(request.storeId)
-                      .thenCompose(
-                          storeRecord ->
-                              storeRecord
-                                  .instance()
-                                  .createDocument(storeRequest)
-                                  .thenApply(
-                                      result ->
-                                          transformResponse(
-                                              request, result, storeRecord.storeId())))
-                      .thenAccept(results::add);
+                  final DocumentStoreRecord documentStore = getDocumentStore(request.storeId);
+                  return documentStore
+                      .instance()
+                      .createDocument(storeRequest)
+                      .handleAsync(
+                          (response, error) ->
+                              results.add(
+                                  transformResponse(
+                                      request, response, error, documentStore.storeId())));
                 })
             .toList();
 
     return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]))
-        .thenApply((ignoredRes) -> results);
+        .handleAsync(
+            (ignoredRes, error) -> {
+              if (error != null) {
+                throw ErrorMapper.mapError(error);
+              }
+              return results;
+            });
   }
 
   public CompletableFuture<DocumentContentResponse> getDocumentContent(
       final String documentId, final String storeId, final String contentHash) {
 
-    return getDocumentStore(storeId)
+    final DocumentStore documentStore = getDocumentStore(storeId).instance();
+    return documentStore
```

Comment:
🙃 Can't we just make the `getDocumentStore` method part of the chain? The variable isn't used anywhere.
```suggestion
    return getDocumentStore(storeId).instance()
```

Reply:
Isn't the variable used in line 124 for `return documentStore.getDocument(documentId);`?

Reply:
😆